test/e2e/pivoting_test.go

```diff
@@ -0,0 +1,263 @@
+			}
+		}
+		return nil
+	}, e2eConfig.GetIntervals(specName, "wait-object-provisioned")...).Should(BeNil())
+
+	By("Check if metal3machines become provisioned.")
+	Eventually(func() error {
+		m3Machines := &capm3.Metal3MachineList{}
+		if err := targetCluster.GetClient().List(ctx, m3Machines, client.InNamespace(namespace)); err != nil {
+			return err
+		}
+		for _, m3Machine := range m3Machines.Items {
+			if !m3Machine.Status.Ready {
+				return errors.New("Metal3Machines cannot be ready")
+			}
+		}
+		return nil
+	}, e2eConfig.GetIntervals(specName, "wait-object-provisioned")...).Should(BeNil())
+
+	By("Check if machines become running.")
+	Eventually(func() error {
+		machines := &clusterv1.MachineList{}
+		if err := targetCluster.GetClient().List(ctx, machines, client.InNamespace(namespace)); err != nil {
+			return err
+		}
+		for _, machine := range machines.Items {
+			if !strings.EqualFold(machine.Status.Phase, "running") { // Case insensitive comparison
+				return errors.New("Machines cannot be in the Running state")
+			}
+		}
+		return nil
+	}, e2eConfig.GetIntervals(specName, "wait-machine-running")...).Should(BeNil())
+
+	By("PASSED!")
+}
+
+func configureIronicConfigmap(isIronicDeployed bool) {
+	bmoPath := "BMOPATH"
+	ironicDataDir := "IRONIC_DATA_DIR"
+	ironicConfigmap := fmt.Sprintf("%s/ironic-deployment/keepalived/ironic_bmo_configmap.env", os.Getenv(bmoPath))
+	newIronicConfigmap := fmt.Sprintf("%s/ironic_bmo_configmap.env", os.Getenv(ironicDataDir))
+	backupIronicConfigmap := fmt.Sprintf("%s/ironic-deployment/keepalived/ironic_bmo_configmap.env.orig", os.Getenv(bmoPath))
+	if isIronicDeployed {
+		cmd := exec.Command("cp", ironicConfigmap, backupIronicConfigmap)
+		err := cmd.Run()
+		Expect(err).To(BeNil(), "Cannot run cp command")
+		cmd = exec.Command("cp", newIronicConfigmap, ironicConfigmap)
+		err = cmd.Run()
+		Expect(err).To(BeNil(), "Cannot run cp command")
+	} else {
+		cmd := exec.Command("mv", backupIronicConfigmap, ironicConfigmap)
+		err := cmd.Run()
+		Expect(err).To(BeNil(), "Cannot run mv command")
+	}
+}
+
+func installIronicBMO(targetCluster framework.ClusterProxy, isIronic, isBMO string) {
+	ironicTLSEnable := "IRONIC_TLS_SETUP"
+	ironicBasicAuth := "IRONIC_BASIC_AUTH"
+	ironicHost := "IRONIC_HOST"
+	bmoPath := "BMOPATH"
+	path := fmt.Sprintf("%s/tools/", os.Getenv(bmoPath))
+	args := []string{
+		isBMO,
+		isIronic,
+		os.Getenv(ironicTLSEnable),
+		os.Getenv(ironicBasicAuth),
+		"true",
+	}
+	env := []string{
+		fmt.Sprintf("IRONIC_HOST=%s", os.Getenv(ironicHost)),
+		fmt.Sprintf("IRONIC_HOST_IP=%s", os.Getenv(ironicHost)),
+		fmt.Sprintf("KUBECTL_ARGS=--kubeconfig=%s", targetCluster.GetKubeconfigPath()),
+		"USER=ubuntu",
+	}
+	cmd := exec.Command("./deploy.sh", args...)
+	cmd.Dir = path
+	cmd.Env = append(env, os.Environ()...)
+
+	err := cmd.Start()
+	Expect(err).To(BeNil(), "Fail to deploy Ironic")
+	err = cmd.Wait()
+	Expect(err).To(BeNil(), "Fail to deploy Ironic")
```

Comment:
```suggestion
	err := cmd.Run()
	Expect(err).To(BeNil(), "Fail to deploy Ironic")
```

Reply:
I will keep this one a bit longer for the sake of debugging. I will change to use Run() in the last time commit